src/background/api.ts

```diff
@@ -57,15 +57,45 @@ function makeHandleRequest(domain: string): Promise<Response> {
   return fetch(requestURL.toString())
 }
 
-export const fetchTwitterHandle = async (tabId: number, domain: string, dispatchBackgroundActions: Dispatchers<BackgroundAction>) => {
-  try {
-    dispatchBackgroundActions.fetchHandleStart({ tabId })
-    const res = await makeHandleRequest(domain)
-    const { twitter_handle } = await res.json()
-    return dispatchBackgroundActions.fetchHandleSuccess({ tabId, domain, handle: twitter_handle })
-  } catch (error) {
-    return dispatchBackgroundActions.fetchHandleFailure({ tabId, domain, error })
-  }
+export const fetchTwitterHandle = async (
+  tabId: number,
+  domain: string,
+  dispatchBackgroundActions: Dispatchers<BackgroundAction>,
+  chrome: typeof global.chrome
+) => {
+  dispatchBackgroundActions.fetchHandleStart({ tabId })
+
+  // We save a cache of the last 50 Twitter handles in the local storage of the browser.
+  chrome.storage.local.get(async result => {
```

Comment:
Wouldn't we want to look up the handles keyed by their domain? 

```typescript
const key = `handle-${domain}`
chrome.storage.local.get([key]), result =>
  console.log(result[key])
  
```

Reply:
Let's split up the bits that are interacting with `chrome.storage` into their own functions

```typescript
function getHandleFromStorage(chrome: typeof global.chrome, domain: string): Promise<Maybe<string>> {
  const key = `handle-${domain}`
  return new Promise(resolve => {
    chrome.storage.local.get([key]), result => resolve(result[key]))
  })
}
```

Reply:
I agree that I should extract the parts handling `chrome.storage`, but I don't agree with creating a new key for each domain. I saved all the handles in one list so we could use it to delete the 51st handle when needed. The reasons I think it's important to save only the last 50 handles are:
a. `chrome.storage` is never deleted. I believe we shouldn't keep even 400KB on our user's computers.
b. Users can change their handles. If we don't check for new handles, the extension will not work for some websites.

Reply:
Those reasons make sense, let's go with that. I think we can do more than 50, but keeping it finite is a good idea.